pinot-common/src/main/java/org/apache/pinot/common/config/TagNameUtils.java

```diff
@@ -101,4 +102,18 @@ public static String getTenantNameFromTag(String tag) {
     }
     return tag;
   }
+
+  public static String getServerTagFromTableConfigAndInstancePartitionsType(TableConfig tableConfig,
+      InstancePartitionsType instancePartitionsType) {
+    switch (instancePartitionsType) {
+      case OFFLINE:
```

Comment:
How does this relate to methods in RealtimeTagConfig class?

Reply:
InstancePartitions is associated with server type (OFFLINE/CONSUMING/COMPLETED). Currently that is configured through the tenant config, and we need this method to extract the server tag from the tenant config.